sirepo/simulation_db.py

```diff
@@ -114,6 +114,11 @@ def assert_sim_db_file_path(basename):
     return basename
 
 
+def cache_dir(dirname, qcall=None):
+    """Returns a named directory within the user's temporary filespace"""
```

Comment:
Names and doc:

```
def cache_dir(basename, qcall):
"""Creates a directory in the user's temporary directory.

Unlike `tmp_dir`, this (possibly new) directory will not be deleted automatically.

Args:
    basename (str): sub-directory to use, need not be unique
    qcall (sirepo.quest.API): request state
Returns:
    py.path: path to newly created directory
"""
```

Note that for new APIs, qcall has to be supplied, i.e. this has to be on the api server or the supervisor.